http_handler.go

```diff
@@ -53,25 +81,44 @@ func (lvl AtomicLevel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		enc.Encode(payload{Level: &current})
 
 	case http.MethodPut:
-		var req payload
-
-		if errmess := func() string {
-			if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-				return fmt.Sprintf("Request body must be well-formed JSON: %v", err)
-			}
-			if req.Level == nil {
-				return "Must specify a logging level."
+		requestedLvl, err := func(body io.Reader) (*zapcore.Level, error) {
+			switch r.Header.Get("Content-Type") {
+			case "application/x-www-form-urlencoded":
+				pld, err := ioutil.ReadAll(body)
+				if err != nil {
+					return nil, err
+				}
+				values, err := url.ParseQuery(string(pld))
+				if err != nil {
+					return nil, err
+				}
+				lvl := values.Get("level")
+				if lvl == "" {
+					return nil, fmt.Errorf("must specify logging level")
+				}
+				var l zapcore.Level
+				if err := l.UnmarshalText([]byte(lvl)); err != nil {
+					return nil, err
+				}
+				return &l, nil
+			default:
+				var pld payload
+				if err := json.NewDecoder(r.Body).Decode(&pld); err != nil {
+					return nil, fmt.Errorf("malformed request body: %v", err)
+				}
+				if pld.Level == nil {
+					return nil, fmt.Errorf("must specify logging level")
+				}
+				return pld.Level, nil
 			}
-			return ""
-		}(); errmess != "" {
+		}(r.Body)
+		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
-			enc.Encode(errorResponse{Error: errmess})
+			enc.Encode(errorResponse{Error: err.Error()})
 			return
 		}
-
-		lvl.SetLevel(*req.Level)
-		enc.Encode(req)
-
+		lvl.SetLevel(*requestedLvl)
+		enc.Encode(payload{Level: requestedLvl})
```

Comment:
Do we want a JSON output even for non-JSON input?

Maybe this is okay and we can add a plain text output later if we need it and the client has an `Accepts: text/plain`?

CC @prashantv

Reply:
If we want to return non-JSON in the response, I would definitively go for content type negotiation through the `Accepts` header.

In any case, I would vote for this to be part of a follow-up PR if it is desired.